compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java

```diff
@@ -144,16 +147,33 @@ void instrumentPackage(BIRPackage pkg) {
             rewriteAsyncInvocations(func, null, pkg);
             rewriteObservableFunctionInvocations(func, pkg);
             if (ENTRY_POINT_MAIN_METHOD_NAME.equals(func.name.value)) {
-                rewriteObservableFunctionBody(func, pkg, null, func.name.value, false, false, true, false);
+                rewriteObservableFunctionBody(func, pkg, null, func.name.value, null, false, false, true, false);
             } else if ((func.flags & Flags.WORKER) == Flags.WORKER) {   // Identifying lambdas generated for workers
-                rewriteObservableFunctionBody(func, pkg, null, func.workerName.value, false, false, false, true);
+                rewriteObservableFunctionBody(func, pkg, null, func.workerName.value, null, false, false, false, true);
             }
         }
         for (BIRTypeDefinition typeDef : pkg.typeDefs) {
             if ((typeDef.flags & Flags.CLASS) != Flags.CLASS && typeDef.type.tag == TypeTags.OBJECT) {
                 continue;
             }
             boolean isService = (typeDef.type.flags & Flags.SERVICE) == Flags.SERVICE;
+            String serviceName = null;
+            if (isService) {
+                for (BIRNode.BIRAnnotationAttachment annotationAttachment : typeDef.annotAttachments) {
+                    if (DISPLAY_ANNOTATION.equals(annotationAttachment.annotTagRef.value)) {
+                        BIRNode.BIRAnnotationRecordValue annotationRecordValue = (BIRNode.BIRAnnotationRecordValue)
+                                annotationAttachment.annotValues.get(0);
+                        Map<String, BIRNode.BIRAnnotationValue> annotationMap =
+                                annotationRecordValue.annotValueEntryMap;
+                        serviceName = ((BIRNode.BIRAnnotationLiteralValue) annotationMap.get("label")).value.toString();
+                        break;
+                    }
+                }
+                if (serviceName == null) {
+                    serviceName = pkg.packageID.orgName.value + "_" + pkg.packageID.name.value + "_svc_" +
```

Comment:
Just as a reference, we discussed about changing this approach in mail "Ballerina Service Names for Distributed Tracing".
Let's merge this for now, hope we can improve the default display name in the future.